aspnetcore/blazor/fundamentals/routing.md

```diff
@@ -19,6 +19,18 @@ This article explains how to manage Blazor app request routing and how to use th
 > [!IMPORTANT]
 > Code examples throughout this article show methods called on `Navigation`, which is an injected <xref:Microsoft.AspNetCore.Components.NavigationManager> in classes and components.
 
+:::moniker range=">= aspnetcore-8.0"
+
+## Static versus interactive routing
+
+*This section applies to Blazor Web Apps.*
+
+If [prerendering isn't disabled](xref:blazor/components/render-modes#prerendering), the Blazor router (`Router` component, `<Router>` in `Routes.razor`) performs static routing to components during static server-side rendering (static SSR). This type of routing is called *static routing*, and the Blazor router is called a *static router* when static routing is taking place.
+
+An interactive render mode can be assigned to the `Routes` component that makes the Blazor router become interactive after static SSR and static routing on the server. For example, `<Routes @rendermode="InteractiveServer" />` assigns the Interactive Server render mode to the `Routes` component. The `Router` component inherits the Interactive Server render mode from the `Routes` component. The router becomes *interactive* after static routing on the server. This type of routing is called *interactive routing*, and the Blazor router is called an *interactive router*. An interactive router can route requests either on the server, *interactive server-side routing*, or on the client, *interactive client-side routing*.
```

Comment:
> after static SSR and static routing on the server

We might be able to just shorten this to "after static SSR" because static server-side routing is just one of things that happens during static SSR, and isn't a separate step. But I could go either way on this and I can see the benefit of explicitly calling out that static server side routing happens before interactivity starts.

> The router becomes *interactive* after static routing on the server

I think "The router becomes *interactive* after static SSR" might be more accurate, because the routing step happens before static SSR completes. It could be possible to interpret the existing wording as meaning "The router becomes interactive _immediately_ after static routing on the server", which wouldn't be accurate.

> An interactive router can route requests either on the server, *interactive server-side routing*, or on the client, *interactive client-side routing*

Do we use this terminology elsewhere? I've been personally thinking of all interactive routing as "client side routing" regardless of whether the router uses Server or WebAssembly interactivity.

Reply:
WRT both terms, we don't use "interactive server-side routing/router" anywhere. We used to use "interactive client routing"/"interactive client-side router" in one sentence in the enhanced nav content, ***but*** we 🔪 that out on that other PR that killed that one confusing paragraph. Now, we don't use any of these terms.

However, it seems unbalanced not to have the terminology available (if it's a thing) because if the app isn't running on the client at all, then there can't be interactive client-side routing, right? Which of the following paradigms matches what the framework does for mixed/Auto render mode apps ...

### There's no such thing as interactive server-side routing beyond static routing

* Static routing/router on the server 👉 Interactive client-side routing/router on the client for WASM-rendered components
* Static routing/router on the server

### Interactive server-side routing occurs after static routing

* Static routing/router on the server 👉 Interactive client-side routing/router on the client for WASM-rendered components
* Static routing/router on the server 👉 Interactive server-side routing/router on the server for server-rendered components

Reply:
The second option you wrote definitely looks more accurate to me, so I'd be happy with going with that.

I've personally been using the term "client-side routing" to mean "any routing that works separately from endpoint routing". For that reason, I would view interactive routing via server interactivity still as a form of "client-side routing". Internal navigations get intercepted, and page content gets generated dynamically by the client (browser) rather than being provided by the server. But maybe others would disagree! @danroth27, do you have an opinion on this?

Reply:
@MackinnonBuck ... I don't think DR will be able to respond due to prior commitments. However, we're close to working this out.

I simplified the text above, distilling it down to just static and interactive routing without getting into the weeds on server/client location. See how that looks, and I'll take another look to see if whatever we say about interactive ***client-side*** routing makes sense based on these remarks that we're working on now.

Reply:
I'm going to go ahead with this. If there's a concern about the final commit, I'll set up a patch PR.